pyomo/core/expr/boolean_value.py

```diff
@@ -0,0 +1,274 @@
+        the state dictionary as control passes up the inheritance
+        hierarchy (using super() calls).  All assignment of state ->
+        object attributes is handled at the last class before 'object',
+        which may -- or may not (thanks to MRO) -- be here.
+        """
+        _base = super(BooleanValue, self)
+        if hasattr(_base, '__setstate__'):
+            return _base.__setstate__(state)
+        else:
+            for key, val in iteritems(state):
+                # Note: per the Python data model docs, we explicitly
+                # set the attribute using object.__setattr__() instead
+                # of setting self.__dict__[key] = val.
+                object.__setattr__(self, key, val)
+
+    def getname(self, fully_qualified=False, name_buffer=None):
+        """
+        If this is a component, return the component's name on the owning
+        block; otherwise return the value converted to a string
+        """
+        _base = super(BooleanValue, self)
+        if hasattr(_base, 'getname'):
+            return _base.getname(fully_qualified, name_buffer)
+        else:
+            return str(type(self))
+
+    @property
+    def name(self):
+        return self.getname(fully_qualified=True)
+
+    @property
+    def local_name(self):
+        return self.getname(fully_qualified=False)
+
+    def cname(self, *args, **kwds):
+        logger.warning(
+            "DEPRECATED: The cname() method has been renamed to getname()." )
+        return self.getname(*args, **kwds)
+
+    def is_constant(self):
+        """Return True if this Logical value is a constant value"""
+        return False
+
+    def is_fixed(self):
+        """Return True if this is a non-constant value that has been fixed"""
+        return False
+
+    def is_parameter_type(self):
+        """Return False unless this class is a parameter object"""
+        return False
+
+    def is_variable_type(self):
+        """Return False unless this class is a variable object"""
+        return False
+
+    def is_potentially_variable(self):
+        """Return True if variables can appear in this expression"""
+        return True
+
+    def is_named_expression_type(self):
+        """Return True if this Logical value is a named expression"""
+        return False
+
+    def is_expression_type(self):
+        """Return True if this Logical value is an expression"""
+        return False
+
+    def is_component_type(self):
+        """Return True if this class is a Pyomo component"""
+        return False
+
+    def is_relational(self):
+        """
+        Return True if this Logical value represents a relational expression.
+        """
+        return True
+
+    def is_indexed(self):
+        """Return True if this Logical value is an indexed object"""
+        return False
+
+    def is_numeric_type(self):
+        """Boolean values are not numeric."""
+        return False
+
+    def equivalent_to(self, other):
```

Comment:
We don't have order-of-operations concerns here?

Reply:
I don't think so?  It is probably useful to list examples for to when we would want to use it (instead of `equivalent_to()`).  As I am typing this, I am separately wondering why you made the method "`equivalent_to`" and not "`iff`"?